src/swaps/blindex/BlindexSwapProvider.js

```diff
@@ -0,0 +1,509 @@
+      return api
+    }
+  }
+
+  getToken(chainId, asset) {
+    const assetData = cryptoassets[asset]
+    if (asset === this.nativeTokenSymbol) {
+      assetData.contractAddress = this.wrbtcAddress
+    }
+
+    return new Token(
+      chainId,
+      assetData.contractAddress,
+      assetData.decimals,
+      assetData.code,
+      assetData.name
+    )
+  }
+
+  getMinimumOutput(outputAmount) {
+    const slippageTolerance = new Percent('200', '10000') // 2%
+    const slippageAdjustedAmountOut = new Fraction(JSBI.BigInt(1))
+      .add(slippageTolerance)
+      .invert()
+      .multiply(outputAmount.quotient).quotient
+    return CurrencyAmount.fromRawAmount(outputAmount.currency, slippageAdjustedAmountOut)
+  }
+
+  /**
+   * Get a quote for the specified parameters
+   * @param {{ network, from, to, amount }} options
+   */
+  // eslint-disable-next-line no-unused-vars
+  async getQuote({ network, from, to, amount }) {
+    if (
+      network != 'mainnet' ||
+      !this.availableTokenSymbols.includes(from) ||
+      !this.availableTokenSymbols.includes(to)
+    ) {
+      return null
+    }
+
+    const fromContractAddress =
+      from === this.nativeTokenSymbol ? this.wrbtcAddress : cryptoassets[from].contractAddress
+    const toContractAddress =
+      to === this.nativeTokenSymbol ? this.wrbtcAddress : cryptoassets[to].contractAddress
+    const bestRouteWithValOut = await this.calcBestRoute(
+      network,
+      to,
+      amount,
+      false,
+      fromContractAddress,
+      toContractAddress
+    )
+    const amountOut_d18 = this.ethersBigNumberToDecimal(bestRouteWithValOut.finalAmount, DECIMALS)
+    return {
+      from,
+      to,
+      fromAmount: currencyToUnit(cryptoassets[from], BN(amount)),
+      toAmount: currencyToUnit(cryptoassets[to], BN(amountOut_d18))
+    }
+  }
+
+  /**
+   * Create a new swap for the given quote
+   * @param {{ network, walletId, quote }} options
+   */
+  // eslint-disable-next-line no-unused-vars
+  async newSwap({ network, walletId, quote }) {
+    const approvalRequired = isERC20(quote.from)
+    const updates = approvalRequired
+      ? await this.approveTokens({ network, walletId, quote })
+      : await this.sendSwap({ network, walletId, quote })
+
+    return {
+      id: uuidv4(),
+      fee: quote.fee,
+      slippage: 200,
+      ...updates
+    }
+  }
+
+  async sendSwap({ network, walletId, quote }) {
+    const txData = await this.buildSwapTx({ network, walletId, quote })
+    const client = this.getClient(network, walletId, quote.from, quote.fromAccountId)
+
+    await this.sendLedgerNotification(quote.fromAccountId, 'Signing required to complete the swap.')
+    const swapTx = await client.chain.sendTransaction(txData)
+
+    return {
+      status: 'WAITING_FOR_SWAP_CONFIRMATIONS',
+      swapTx,
+      swapTxHash: swapTx.hash
+    }
+  }
+
+  getChecksumAddressFromAddress(address) {
+    return ethers.utils.getAddress(address.toLowerCase())
```

Comment:
Do we need to use `toLowerCase` here?
I believe `ethers.utils.getAddress` makes sure we get the right address even without it

Reply:
We need.
If given address is mixed-cased then getAddress has another check to verify if the address is equal to checksum address.
If its not it logs an error.
The config which I have copied from frontent environment.prod does not have checksummed addresses apparently.
![image](https://user-images.githubusercontent.com/91085805/161044013-b4c666ff-4c8d-44f9-89ab-34ff3d6bb0f1.png)